railties/test/commands/dev_cache_test.rb

```diff
@@ -0,0 +1,32 @@
+require_relative '../isolation/abstract_unit'
+
+module CommandsTests
+  class DevCacheTest < ActiveSupport::TestCase
+    include ActiveSupport::Testing::Isolation
+
+    def setup
+      build_app
+    end
+
+    def teardown
+      teardown_app
+    end
+
+    test 'dev:cache creates file and outputs message' do
+      Dir.chdir(app_path) do
+        output = `rails dev:cache`
+        assert File.exist?('tmp/caching-dev.txt')
+        assert_match(%r{Development mode is now being cached}, output)
+      end
+    end
+
+    test 'dev:cache deletes file and outputs message' do
+      Dir.chdir(app_path) do
+        output = `rails dev:cache`        
```

Comment:
Cleared it up with some comments: https://github.com/rails/rails/commit/2af9c080799c25a0bb4f6f6b8d2685b6c0e47299 :grin: